test/puppetlabs/puppetdb/scf/storage_test.clj

```diff
@@ -511,17 +511,11 @@
                                " JOIN catalogs ON catalogs.id = latest_catalogs.catalog_id"
                                " WHERE certnames.certname = 'basic.catalogs.com'")]))))
 
-  (testing "garbage collection deletes old catalogs"
-    (jdbc/with-transacted-connection *db*
-      (delete-old-unassociated-catalogs! (-> 1 days ago)))
-    (is (= [{:count 1}]
-           (query-to-vec ["SELECT COUNT(*) FROM catalogs"]))))
-
-  (testing "garbage collection leaves the newest catalogs"
-    (jdbc/with-transacted-connection *db*
-      (delete-old-unassociated-catalogs! (now)))
-    (is (= [{:count 1}]
-           (query-to-vec ["SELECT COUNT(*) FROM catalogs"])))))
+  (testing "only stores up to three catalogs a certname"
+    (store-catalog! (assoc catalog :producer_timestamp (-> 3 days ago)) (now))
+    (store-catalog! (assoc catalog :producer_timestamp (-> 4 days ago)) (now))
+    (store-catalog! (assoc catalog :producer_timestamp (-> 5 days ago)) (now))
+    (is (= [{:count 3}] (query-to-vec ["SELECT COUNT(*) FROM catalogs"])))))
```

Comment:
we should have a test case for what happens when we add a fourth. ... unless I've misread this one and that's what it's actually doing.